### PR TITLE
Adding tests

### PR DESCRIPTION
**Pull request by diyahir**
- Adding bls tests
- Adding u256 conversion tests
- Fixing u256 bug
- typos

### CODE REVIEW

File: crates/crypto/bls/src/attestation.rs [hunk 1]
```diff
@@ -185,3 +185,78 @@ impl G1Point {
...
+
+    #[tokio::test]
+    async fn test_keypair_generation() {
+        let mut rng = thread_rng();
+        let private_key = Fr::rand(&mut rng);
+        let keypair = KeyPair::new(private_key).unwrap();
+        let pub_key = keypair.get_pub_key_g1();
+
+        // Check that the public key is not zero
+        assert_ne!(pub_key, G1Projective::zero());
```
Comment (supernovahs) [inline crates/crypto/bls/src/attestation.rs:203]: curious to know how asserting the pub key as not zero helps us. Is this related to any arkworks library thing?

File: crates/crypto/bls/src/attestation.rs [hunk 2]
```diff
@@ -185,3 +185,78 @@ impl G1Point {
...
+        let private_key = Fr::rand(&mut rng);
+        let keypair = KeyPair::new(private_key).unwrap();
+
+        let message = [0u8; 32];
+        let msg_hash = hash_to_g1(&message);
+
+        let signature = keypair.sign_hashes_to_curve_message(msg_hash.into());
+
+        // Check that the signature is not zero
+        assert_ne!(signature.sig(), G1Projective::zero());
```
Comment (supernovahs) [inline crates/crypto/bls/src/attestation.rs:218]: same as above

File: crates/crypto/bls/src/attestation.rs [hunk 3]
```diff
@@ -185,3 +185,78 @@ impl G1Point {
...
+        // generate a random message
+        let mut message = [0u8; 32];
+        rng.fill_bytes(&mut message);
+
+        let msg_hash = hash_to_g1(&message);
+
+        let signature = keypair.sign_hashes_to_curve_message(msg_hash.into());
+
+        // Check that the signature is not zero
+        assert_ne!(signature.sig(), G1Projective::zero());
```
Comment (supernovahs) [inline crates/crypto/bls/src/attestation.rs:236]: same

File: crates/crypto/bls/src/attestation.rs [hunk 4]
```diff
@@ -185,3 +185,78 @@ impl G1Point {
...
+
+        let mut message = [0u8; 32];
+        rng.fill_bytes(&mut message);
+        
+        let msg_hash = hash_to_g1(&message);
+
+        let signature = keypair.sign_hashes_to_curve_message(msg_hash.into());
+
+        // Check that the signature is not zero
+        assert_ne!(signature.sig(), G1Projective::zero());
```
Comment (supernovahs) [inline crates/crypto/bls/src/attestation.rs:256]: same

**Review by supernovahs — COMMENTED** [review]
thanks
nightly fmt pending apart from above nit. then lgtm

**Comment by supernovahs** [comment]
hey @diyahir , do you have the bandwith right now to clear the nits?Else  I will be happy to take this over the finish line

**Comment by diyahir** [comment]
> hey @diyahir , do you have the bandwith right now to clear the nits?Else I will be happy to take this over the finish line

Ah rip sorry, yeah I am caught it in some other development right now -- For the non-zero check, I saw it in some other bls tests, not entirely sure what it does I believe it has to do with cryptography breaking if zero pubkey, but that's purely speculation

**Comment by supernovahs** [comment]
thanks superseded by #10
